streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java

```diff
@@ -36,10 +39,22 @@
 
     boolean commitNeeded();
 
+    default boolean commitRequested() {
```

Comment:
This is a meta question: I think we should really maintain functional methods that would be used for all types of tasks here, otherwise let each extended class have their own function.